test/extensions/wasm/wasm_test.cc

```diff
@@ -31,8 +32,10 @@ TEST(WasmTest, Logging) {
   Stats::IsolatedStoreImpl stats_store;
   Api::ApiPtr api = Api::createApiForTest(stats_store);
   Event::SimulatedTimeSystem time_system;
+  Upstream::MockClusterManager cluster_manager;
   Event::DispatcherImpl dispatcher(time_system, *api);
-  auto wasm = std::make_unique<Extensions::Common::Wasm::Wasm>("envoy.wasm.vm.wavm", "");
+  auto wasm = std::make_unique<Extensions::Common::Wasm::Wasm>("envoy.wasm.vm.wavm", "", "",
```

Comment:
It's hard for me to reason about if this var `wasm` inovokes shared_from_this(). Could you just use `shared_ptr<Wasm>`? Basically any unique_ptr<Wasm> could lead to an undefined behavior due to shared_from_this()

Reply:
The idiom for services (as opposed to filters) is to store the resulting object as a unique_ptr<>.  However, I agree that in this case it is better to break with that idiom for safety and consistency.  I'll update the to use only shared_ptr.

Reply:
See the latest commit.